### PR TITLE
Standardize intallation enum documentation format

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Some of the enums didn't list the valid values, resulting in docs that were incomplete, like this: https://docs.projectcalico.org/reference/installation/api#operator.tigera.io/v1.BGPOption

This PR standardizes on a format for enums that should render nicely in the docs. 

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by tmjd** [comment]
Do we have any in this format already that I can see what this looks like in the API reference?

File: api/v1/installation_types.go [hunk 1]
```diff
@@ -135,7 +135,9 @@ type PreferredNodeAffinity struct {
 	PreferredDuringSchedulingIgnoredDuringExecution []v1.PreferredSchedulingTerm `json:"preferredDuringSchedulingIgnoredDuringExecution,omitempty"`
 }
 
-// ComponentName CRD enum
+// ComponentName represents a single component.
+//
+// One of: Node, Typha, KubeControllers, ECKOperator
```
Comment (tmjd) [inline api/v1/installation_types.go:140]: This is wrong now because ECKOperator is now in its own type.

Comment (caseydavenport) [reply api/v1/installation_types.go:140]: According to the enum itself it's still valid (see line 147 below) - should I remove that?

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by caseydavenport** [comment]
> Do we have any in this format already that I can see what this looks like in the API reference?

Nope, nothing yet. But it renders basically as you'd expect (I built it locally).

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**